server/integration/testsuite/src/test/java/org/infinispan/server/test/task/DistributedAuthServerTaskIT.java

```diff
@@ -0,0 +1,121 @@
+import org.infinispan.tasks.ServerTask;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests in distributed mode the server task execution in case if authentication is required.
+ *
+ * @author amanukya
+ */
+@RunWith(Arquillian.class)
+@Category({Task.class})
+@WithRunningServer({@RunningServer(name="hotrodAuthClustered"), @RunningServer(name = "hotrodAuthClustered-2")})
+public class DistributedAuthServerTaskIT {
+    @InfinispanResource("hotrodAuthClustered")
+    RemoteInfinispanServer server1;
+
+    @InfinispanResource("hotrodAuthClustered-2")
+    RemoteInfinispanServer server2;
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @BeforeClass
+    public static void before() throws Exception {
+        String[] serverDirs = new String[]{System.getProperty("server1.dist"), System.getProperty("server2.dist")};
+
+        JavaArchive jar = ShrinkWrap.create(JavaArchive.class);
+        jar.addClass(DistributedAuthServerTask.class);
+        jar.addClass(LocalAuthTestServerTask.class);
+        jar.addAsServiceProvider(ServerTask.class, DistributedAuthServerTask.class, LocalAuthTestServerTask.class);
+
+        for (String serverDir : serverDirs) {
+            File f = new File(serverDir, "/standalone/deployments/custom-distributed-task-with-auth.jar");
+            f.deleteOnExit();
+            jar.as(ZipExporter.class).exportTo(f, true);
+        }
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldRunLocalAuthTest() throws Exception {
+        SaslConfigurationBuilder config = new SaslConfigurationBuilder("DIGEST-MD5");
+        config.forIspnServer(server1).withServerName("node0");
+        config.forCredentials(HotRodSaslAuthTestBase.SUPERVISOR_LOGIN, HotRodSaslAuthTestBase.SUPERVISOR_PASSWD);
+        RemoteCacheManager rcm = new RemoteCacheManager(config.build(), true);
+        RemoteCache remoteCache = rcm.getCache(LocalAuthTestServerTask.CACHE_NAME);
+
+        String result = (String) remoteCache.execute(LocalAuthTestServerTask.NAME, Collections.emptyMap());
+        assertEquals(LocalAuthTestServerTask.EXECUTED_VALUE, result);
+        assertEquals(true, remoteCache.get(LocalAuthTestServerTask.KEY));
+
+        rcm.stop();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldRunDistAuthTest() throws Exception {
```

Comment:
It seems there is a failure: http://ci.infinispan.org/viewLog.html?buildId=37568&buildTypeId=bt9&guest=1#testNameId-1517987404531832027